### PR TITLE
feat: add DO example

### PR DESCRIPTION
**Pull request by rsmitty**
This PR adds the digitalocean example for cluster creation w/ terraform. This PR will depend on the upstream DO provider to cut a new release.

Signed-off-by: Spencer Smith <spencer.smith@talos-systems.com>

### CODE REVIEW

**Comment by rsmitty** [comment]
This will have to hang out here until 1.26.1 is cut after https://github.com/digitalocean/terraform-provider-digitalocean/pull/931 got merged.

**Review by frezbo — APPROVED** [review]

File: examples/terraform/digitalocean/variables.tf [hunk 1]
```diff
@@ -0,0 +1,41 @@
+variable "cluster_name" {
+  description = "Name of cluster"
+  type        = string
+  default     = "talos-do"
+}
+
+variable "num_control_plane" {
+  description = "Number of control plane nodes to create"
+  type        = number
+  default     = 1
```
Comment (alubbock) [inline examples/terraform/digitalocean/variables.tf:10]: The README states that 3 control plane nodes will be created, but the default specified here is 1.

Comment (rsmitty) [reply examples/terraform/digitalocean/variables.tf]: good call, thank you! fixed in amended commit.

**Review by frezbo — APPROVED** [review]
Also need a PR to the TF provider repo to add in this example. I wonder if we just point to this repo rather than explicitly call out cloud examples :sweat_smile:

**Pull request merged**